src/app/modules/submission/email-submission/__tests__/email-submission.routes.spec.ts

```diff
@@ -97,6 +109,395 @@ describe('email-submission.routes', () => {
+              {
+                ...MOCK_ATTACHMENT_RESPONSE,
+                content: MOCK_ATTACHMENT_RESPONSE.content.toString('binary'),
+              },
+            ],
+          }),
+        )
+        .query({ captchaResponse: 'null' })
+
+      expect(response.status).toBe(200)
+      expect(response.body).toEqual({
+        message: 'Form submission successful.',
+        submissionId: expect.any(String),
+      })
+    })
+
+    it('should return 200 when response has isHeader key', async () => {
+      const { form } = await dbHandler.insertEmailForm({
+        formOptions: {
+          hasCaptcha: false,
+          status: Status.Public,
+          form_fields: [MOCK_SECTION_FIELD],
+        },
+      })
+
+      const response = await request
+        .post(`${SUBMISSIONS_ENDPT_BASE}/${form._id}`)
+        .field(
+          'body',
+          JSON.stringify({
+            isPreview: false,
+            responses: [{ ...MOCK_SECTION_RESPONSE, isHeader: true }],
+          }),
+        )
+        .query({ captchaResponse: 'null' })
+
+      expect(response.status).toBe(200)
+      expect(response.body).toEqual({
+        message: 'Form submission successful.',
+        submissionId: expect.any(String),
+      })
+    })
+
+    it('should return 200 when signature is empty string for optional verified field', async () => {
+      const { form } = await dbHandler.insertEmailForm({
+        formOptions: {
+          hasCaptcha: false,
+          status: Status.Public,
+          form_fields: [MOCK_OPTIONAL_VERIFIED_FIELD],
+        },
+      })
+
+      const response = await request
+        .post(`${SUBMISSIONS_ENDPT_BASE}/${form._id}`)
+        .field(
+          'body',
+          JSON.stringify({
+            isPreview: false,
+            responses: [{ ...MOCK_OPTIONAL_VERIFIED_RESPONSE, signature: '' }],
+          }),
+        )
+        .query({ captchaResponse: 'null' })
+
+      expect(response.status).toBe(200)
+      expect(response.body).toEqual({
+        message: 'Form submission successful.',
+        submissionId: expect.any(String),
+      })
+    })
+
+    it('should return 200 when response has answerArray and no answer', async () => {
+      const { form } = await dbHandler.insertEmailForm({
+        formOptions: {
+          hasCaptcha: false,
+          status: Status.Public,
+          form_fields: [MOCK_CHECKBOX_FIELD],
+        },
+      })
+
+      const response = await request
+        .post(`${SUBMISSIONS_ENDPT_BASE}/${form._id}`)
+        .field(
+          'body',
+          JSON.stringify({
+            isPreview: false,
+            responses: [MOCK_CHECKBOX_RESPONSE],
+          }),
+        )
+        .query({ captchaResponse: 'null' })
+
+      expect(response.status).toBe(200)
+      expect(response.body).toEqual({
+        message: 'Form submission successful.',
+        submissionId: expect.any(String),
+      })
+    })
+
+    it('should return 400 when isPreview key is missing', async () => {
```

Comment:
wondering if we really need to check for isPreview in Joi validation, since it's not used in the backend at all

Reply:
yeah, I'll deprecate it separately